ResourceRequest timed out

Node: 6.9.1
configuration: 
```
pool: {
    max: 100,
    min: 1,
    idleTimeoutMillis: 15000
}
```


I am getting this error when there is less traffic or there is gap between requests.


```
error:  { TimeoutError: ResourceRequest timed out
        at ResourceRequest._fireTimeout (/home/munir/node-workspace/server/node_modules/generic-pool/lib/ResourceRequest.js:62:17)
        at Timeout.bound (/home/munir/node-workspace/server/node_modules/generic-pool/lib/ResourceRequest.js:8:15)
        at ontimeout (timers.js:365:14)
        at tryOnTimeout (timers.js:237:5)
        at Timer.listOnTimeout (timers.js:207:5) name: 'TimeoutError' }
```

@willmorgan 

Here is the stack trace.

```log
[2018-07-26T09:12:41.690] [ERROR] default - Database connection pool error { error: 
   { ConnectionError: Failed to connect to 1.1.1.1:1433 - connect EHOSTUNREACH 1.1.1.1:1433
    at Connection.tedious.once.err (C:\projects\host\node_modules\mssql\lib\tedious.js:216:17)
    at Object.onceWrapper (events.js:255:19)
    at Connection.emit (events.js:160:13)
    at Connection.socketError (C:\projects\host\node_modules\tedious\lib\connection.js:1016:14)
    at C:\projects\host\node_modules\tedious\lib\connection.js:861:25
    at Socket.onError (C:\projects\host\node_modules\tedious\lib\connector.js:49:9)
    at Socket.emit (events.js:160:13)
    at emitErrorNT (internal/streams/destroy.js:64:8)
    at process._tickCallback (internal/process/next_tick.js:152:19)
     code: 'ESOCKET',
     originalError: 
      { ConnectionError: Failed to connect to 1.1.1.1:1433 - connect EHOSTUNREACH 1.1.1.1:1433
    at ConnectionError (C:\projects\host\node_modules\tedious\lib\errors.js:12:12)
    at Connection.socketError (C:\projects\host\node_modules\tedious\lib\connection.js:1016:30)
    at C:\projects\host\node_modules\tedious\lib\connection.js:861:25
    at Socket.onError (C:\projects\host\node_modules\tedious\lib\connector.js:49:9)
    at Socket.emit (events.js:160:13)
    at emitErrorNT (internal/streams/destroy.js:64:8)
    at process._tickCallback (internal/process/next_tick.js:152:19)
        message: 'Failed to connect to 1.1.1.1:1433 - connect EHOSTUNREACH 1.1.1.1:1433',
        code: 'ESOCKET' },
     name: 'ConnectionError' } }
```

--- a/lib/tedious.js
+++ b/lib/tedious.js
@@ -231,8 +231,10 @@ class ConnectionPool extends base.ConnectionPool {
       IDS.add(tedious, 'Connection')
       debug('pool(%d): connection #%d created', IDS.get(this), IDS.get(tedious))
       debug('connection(%d): establishing', IDS.get(tedious))
+      const emittedConnectEvent = false
 
       tedious.once('connect', err => {
+        emittedConnectEvent = true
         if (err) {
           err = new base.ConnectionError(err)
           return reject(err)
@@ -241,6 +243,12 @@ class ConnectionPool extends base.ConnectionPool {
         debug('connection(%d): established', IDS.get(tedious))
         resolve(tedious)
       })
+
+      setTimeout(() => {
+        if (!emittedConnectEvent) {
+          reject(new base.ConnectionError('Tedious connection timed out after 10 seconds'))
+        }
+      }, 10 * 1000)
 
       tedious.on('error', err => {
         if (err.code === 'ESOCKET') {

--- a/lib/tedious.js
+++ b/lib/tedious.js
@@ -231,10 +231,8 @@ class ConnectionPool extends base.ConnectionPool {
       IDS.add(tedious, 'Connection')
       debug('pool(%d): connection #%d created', IDS.get(this), IDS.get(tedious))
       debug('connection(%d): establishing', IDS.get(tedious))
-      let emittedConnectEvent = false
 
       tedious.once('connect', err => {
-        emittedConnectEvent = true
         if (err) {
           err = new base.ConnectionError(err)
           return reject(err)
@@ -243,12 +241,6 @@ class ConnectionPool extends base.ConnectionPool {
         debug('connection(%d): established', IDS.get(tedious))
         resolve(tedious)
       })
-
-      setTimeout(() => {
-        if (!emittedConnectEvent) {
-          reject(new base.ConnectionError('Tedious connection timed out after options.connectTimeout'))
-        }
-      }, cfg.options.connectTimeout)
 
       tedious.on('error', err => {
         if (err.code === 'ESOCKET') {

--- a/lib/tedious.js
+++ b/lib/tedious.js
@@ -246,9 +246,9 @@ class ConnectionPool extends base.ConnectionPool {
 
       setTimeout(() => {
         if (!emittedConnectEvent) {
-          reject(new base.ConnectionError('Tedious connection timed out after 10 seconds'))
+          reject(new base.ConnectionError('Tedious connection timed out after options.connectTimeout'))
         }
-      }, 10 * 1000)
+      }, cfg.options.connectTimeout)
 
       tedious.on('error', err => {
         if (err.code === 'ESOCKET') {

--- a/lib/tedious.js
+++ b/lib/tedious.js
@@ -253,10 +253,12 @@ class ConnectionPool extends base.ConnectionPool {
       tedious.on('error', err => {
         if (err.code === 'ESOCKET') {
           tedious.hasError = true
+          reject(err)
           return
         }
 
         this.emit('error', err)
+        reject(err)
       })
 
       if (this.config.debug) {

--- a/lib/tedious.js
+++ b/lib/tedious.js
@@ -231,7 +231,7 @@ class ConnectionPool extends base.ConnectionPool {
       IDS.add(tedious, 'Connection')
       debug('pool(%d): connection #%d created', IDS.get(this), IDS.get(tedious))
       debug('connection(%d): establishing', IDS.get(tedious))
-      const emittedConnectEvent = false
+      let emittedConnectEvent = false
 
       tedious.once('connect', err => {
         emittedConnectEvent = true

